### ISSUE TITLE
G loss falls to 0 and D loss stays at .5

### ISSUE THREAD

**Issue by cppietime** [HyperGAN/HyperGAN#177]
Running the HyperGAN cli on a set of images (no classes) with the following command:
`hypergan train folder/ -s 64x64x3 -b 32 -f png -c model --resize --save_every 100`
Resulted in the network producing fuzzy images that vaguely resembled the training data for a while, with loss values of about .02 for d and .4 for g. After several thousand epochs, the g-loss began to rise, and before long, the network produced the same meaningless images over and over again with a g-loss of 0 and d-loss of exactly .5. Then it just stays there indefinitely. Is there a way to resolve this issue so that it does not end up at an inescapable state as described above?

**Comment by qwazwsx** [comment]
This is whats called 'mode collapse' 

you can read more [here](http://aiden.nibali.org/blog/2017-01-18-mode-collapse-gans/)

in a **very** oversimplified way:

*GANS in general*

There are two networks, the generator and the discriminator. The generators job is to take random noise and make something that is close to what you are training it on. The discriminators job is to look at either a real image (something from your training dataset) or an image generated by the generator. If it thinks the image is real while it is actually a generated image, the generator is happy and learns to make more images like that, while the generator is sad and learns from its mistake. But if the discriminator gets a generated image and notices that its a fake, the generator is punished and the discriminator is happy. It can be compared to a counterfeiter trying to make fake bills (the generator) and a detective trying to figure out what bills are fake or not (the discriminator). The generator continuously making fakes and figuring out which ones worked and which ones didn't. And the detective constantly analyzing new bills trying to find fakes. 

now on to mode colapse

Lets say you are training hypergan to generate pictures of Giraffes.

After a while the generator figures out a weakness in the discriminator; if the image it generated has a long neck, the discriminator will think its a real image. So, the generator starts focusing all its energy on long necks. Soon the discriminator gets wise and realizes that its being tricked. So it learns that if it has a long neck its a fake. The discriminator then starts focusing all of its energy on finding images with long necks. Then the generator catches on to the discriminators NEW weakness, so it starts outputting images with SHORT necks. The discriminator thinks images with short necks must be real because only fake images have long necks right?. Repeat this process over and over until you either are generating random colors and shapes, or the same image over and over.


In reality it's much more complicated and relies on things more abstract than neck length. 

I would recommend trying out different configurations and see what works for you. Getting it to train is one of the main difficulties of GANs

**Comment by martyn** [comment]
Please try again with the new default in hypergan 0.10
